test/test_e2e.rb

```diff
@@ -104,4 +104,24 @@ def test_spec_file_splitting
       "./spec/fast_spec.rb",
     ], queue)
   end
+
+  def test_suite_with_failures_and_fail_fast
+    queue = exec_build("failing_suite", "--fail-fast 1")
+
+    assert queue.fail_fast == 1
+    assert queue.unprocessed_jobs.length <= 2
+    assert queue.build_failed_fast?
+    refute queue.build_successful?
+    assert queue.example_failures.length + queue.non_example_errors.length ==
+           queue.fail_fast
```

Comment:
Let's be more concrete here, since this is a test and we want to know what we're testing for at a glance. The given suite doesn't contain an error so we shouldn't test about it:

```ruby
assert queue.example_failures.length == queue.fail_fast
```